apps/web/app/api/epub/route.ts

```diff
@@ -0,0 +1,65 @@
+import { inArray } from "drizzle-orm";
+import epub, { Chapter } from "epub-gen-memory";
+
+import { db } from "@hoarder/db";
+import { bookmarkLinks } from "@hoarder/db/schema";
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const assetIds = searchParams.getAll("assetId");
+
+  if (!assetIds || assetIds.length === 0) {
+    return new Response("", {
+      status: 404,
+    });
+  }
+
+  const bookmarkInformation = await db
+    .select({
+      title: bookmarkLinks.title,
+      htmlContent: bookmarkLinks.htmlContent,
+    })
+    .from(bookmarkLinks)
+    .where(inArray(bookmarkLinks.id, assetIds));
+
+  if (!bookmarkInformation || bookmarkInformation.length === 0) {
+    return new Response("", {
+      status: 404,
+    });
+  }
+
+  const chapters = bookmarkInformation.map((information) => {
+    return {
+      content: information.htmlContent ?? "",
+      title: information.title ?? "",
+    };
+  });
+
+  const title = getTitle(chapters);
+
+  const generatedEpub = await epub(title, chapters);
```

Comment:
I think we should also remove the table of contents when it's only a single bookmark

Reply:
Seems like this is actually a requirement for epubs: https://ebooks.stackexchange.com/questions/7209/epub-without-a-table-of-contents-is-it-possible
I had a look at the library and it is hardcoded to generate it. I then tried an empty template and it does not work.

Reply:
was able to fix that by updating the template